services/web/server/src/simcore_service_webserver/projects/_workspaces_api.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+from aiohttp import web
+from models_library.products import ProductName
+from models_library.projects import ProjectID
+from models_library.users import UserID
+from models_library.workspaces import WorkspaceID
+
+from ..projects._access_rights_api import get_user_project_access_rights
+from ..users.api import get_user
+from ..workspaces.api import check_user_workspace_access
+from . import _folders_db as project_to_folders_db
+from . import _groups_db as project_groups_db
+from .db import APP_PROJECT_DBAPI, ProjectDBAPI
+from .exceptions import ProjectInvalidRightsError
+
+_logger = logging.getLogger(__name__)
+
+
+async def move_project_into_workspace(
+    app: web.Application,
+    *,
+    user_id: UserID,
+    project_id: ProjectID,
+    workspace_id: WorkspaceID | None,
+    product_name: ProductName,
+) -> None:
+    project_api: ProjectDBAPI = app[APP_PROJECT_DBAPI]
+
+    # 1. User needs to have delete permission on project
+    project_access_rights = await get_user_project_access_rights(
+        app, project_id=project_id, user_id=user_id, product_name=product_name
+    )
+    if project_access_rights.delete is False:
+        raise ProjectInvalidRightsError(user_id=user_id, project_uuid=project_id)
+
+    # 2. User needs to have write permission on workspace
```

Comment:
... will say nothing

Reply:
I know still in my todo list, I will do it ones everywhere and properly ;)